src/org/jetbrains/java/decompiler/modules/decompiler/sforms/SSAConstructorSparseEx.java

```diff
@@ -120,174 +114,190 @@ private void ssaStatements(DirectGraph dgraph, HashSet<String> updated, StructMe
     }
   }
 
-  private void processExprent(DirectNode node, Exprent expr, SFormsFastMapDirect[] varmaparr) {
+  private void processExprent(DirectNode node, Exprent expr, VarMapHolder varmaps) {
```

Comment:
I think this method needs some docs as to what it's doing, and some minor docs for each of the switch branches